crates/assemble/src/npm/generators.rs

```diff
@@ -16,10 +16,11 @@ pub fn module_types(
 ) -> String {
     let mut w = String::with_capacity(4096);
 
-    let doc_mapper = build_mapper(compiled, imports, &collection.schema, true);
+    let w_mapper = build_mapper(compiled, imports, &collection.write_schema, false);
+    let r_mapper = build_mapper(compiled, imports, &collection.read_schema, true);
```

Comment:
So the logic in this function generates types `SourceDocument` and `OutputDocument` but then only the `SourceDocuments` are re-exported as `<Transform>Source` later on, and the `OutputDocument` types are referenced directly in the actual derivation code. Do I have that right?

Reply:
Yep, that's right.

And in the combined-schema case we generate `Document` which aliases to `SourceDocument` & `OutputDocument`, which has the convenient property of not breaking existing derivations which import that `Document` type.